### PR TITLE
[Enhancement]support paimon version 0.5

### PR DESCRIPTION
**Pull request by wg1026688210**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by miomiocat — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|0|1|00.00%|[210]|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|1|00.00%|[140]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|2|3|66.67%|[145]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|6|7|85.71%|[215]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|2|3|66.67%|[145]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|6|7|85.71%|[215]|

File: fe/fe-core/src/test/java/com/starrocks/connector/paimon/PaimonMetadataTest.java [hunk 1]
```diff
@@ -85,10 +85,17 @@ public void setUp() {
...
+        this.splits.add(DataSplit.builder().withSnapshot(1L).withPartition(row1).withBucket(1).withDataFiles(meta1)
+                .isStreaming(false).build());
+        this.splits.add(DataSplit.builder().withSnapshot(1L).withPartition(row2).withBucket(1).withDataFiles(meta2)
+                .isStreaming(false).build());
     }
 
+    @Test
+    public void testRowCount() {
+        long rowCount = metadata.getRowCount(splits);
+        Assert.assertTrue(rowCount > 0);
```
Comment (miomiocat) [inline fe/fe-core/src/test/java/com/starrocks/connector/paimon/PaimonMetadataTest.java:97]: `rowCount==2`

File: fe/fe-core/src/test/java/com/starrocks/planner/PaimonScanNodeTest.java [hunk 2]
```diff
@@ -48,4 +60,28 @@ public void testInit(@Mocked GlobalStateMgr globalStateMgr,
...
+                1, 1, 1, DUMMY_LEVEL));
+
+        List<DataFileMeta> meta2 = new ArrayList<>();
+        meta2.add(new DataFileMeta("file3", 100, 400, EMPTY_MIN_KEY, EMPTY_MAX_KEY, EMPTY_KEY_STATS, null,
+                1, 1, 1, DUMMY_LEVEL));
+
+        DataSplit split = DataSplit.builder().withSnapshot(1L).withPartition(row1).withBucket(1).withDataFiles(meta1)
+                .isStreaming(false).build();
+        long totalFileLength = PaimonScanNode.getTotalFileLength(split);
+        Assert.assertTrue(totalFileLength > 0);
```
Comment (miomiocat) [inline fe/fe-core/src/test/java/com/starrocks/planner/PaimonScanNodeTest.java:85]: `totalFileLength==200`

File: fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java [hunk 3]
```diff
@@ -154,6 +155,12 @@ private void addScanRangeLocations(DataSplit split, String predicateInfo) {
         scanRangeLocationsList.add(scanRangeLocations);
     }
 
+    @VisibleForTesting
+    static long getTotalFileLength(DataSplit split) {
```
Comment (miomiocat) [inline fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java:159]: remove `static` and `@VisibleForTesting`

File: fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java [hunk 4]
```diff
@@ -225,6 +222,16 @@ public Statistics getTableStatistics(OptimizerContext session,
         return builder.build();
     }
 
+    @VisibleForTesting
```
Comment (miomiocat) [inline fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java:225]: remove `VisibleForTesting`

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|2|3|66.67%|[145]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|6|7|85.71%|[215]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|2|3|66.67%|[145]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|6|7|85.71%|[215]|

**Review by miomiocat — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|2|3|66.67%|[146]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|6|7|85.71%|[214]|

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**